### ISSUE TITLE
Possible PHP Warning trying to reference to a property of a non object in class.llms.comments.php

### ISSUE THREAD

**Issue by eri-trabiccolo** [gocodebox/lifterlms#1049]
Ok the this issue title is weird, I see. The "problem" is that we don't have a real way to reproduce the issue one of our customer reported if not forcing it.
@nrherron92 and I bet on a conflicting plugin, although I think is something we should handle better.
Ah and of course, the visible issue, is that our order notes (so try to have some order notes too) etc were counted as comments.

### Reproduction Steps
(force)
+ add the following somewhere
```php
delete_transient( 'llms_count_comments' );
add_filter( 'wp_count_comments', '__return_empty_array', 30 );
```
+ check the logs (check the comments count e.g. in dashboard -> Comments)

### Expected Behavior

+ nothing (comments count should not change from before forcing the issue)

### Actual Behavior

+ php notices (comments count is bigger than before forcing the issue)

### Error Messages / Logs

```
[09-Mar-2020 15:16:31 UTC] PHP Warning:  Attempt to assign property 'total_comments' of non-object in /srv/www/wordpress-one/public_html/wp-content/plugins/lifterlms/includes/class.llms.comments.php on line 137
[09-Mar-2020 15:16:31 UTC] PHP Warning:  Attempt to assign property 'approved' of non-object in /srv/www/wordpress-one/public_html/wp-content/plugins/lifterlms/includes/class.llms.comments.php on line 142
```

### System Information

Below the system info of the user who reported it.

<details>
<summary>System Report</summary>
  
```
Wordpress
-------------------------------------------

Home Url: https://{REDACTED}
Site Url: https://{REDACTED}
Login Url: https://{REDACTED}/wp-login.php
Version: 5.3.2
Debug Mode: No
Debug Log: No
Debug Display: Yes
Locale: en_US
Multisite: No
Page For Posts: Not Set
Page On Front: Home (#19) [https://{REDACTED}/]
Permalink Structure: /%postname%/
Show On Front: page
Wp Cron: Yes


Settings
-------------------------------------------

Version: 3.37.11
Db Version: 3.37.11
Course Catalog: Bible Study Menu (#150) [https://{REDACTED}/courses/]
Membership Catalog: Become a Member (#151) [https://{REDACTED}/memberships/]
Student Dashboard: My Mama’s Refill Account (#153) [https://{REDACTED}/my-courses/]
Checkout Page: Ready to start studying with us? (#152) [https://{REDACTED}/join/]
Course Catalog Per Page: -1
Course Catalog Sorting: date,DESC
Membership Catalog Per Page: -1
Membership Catalog Sorting: menu_order,ASC
Site Membership: Not Set
Courses Endpoint: my-studies
Edit Endpoint: edit-account
Lost Password Endpoint: lost-password
Vouchers Endpoint: redeem-voucher
Autogenerate Username: no
Password Strength Meter: no
Minimum Password Strength: strong
Terms Required: no
Terms Page: Not Set
Checkout Names: required
Checkout Address: required
Checkout Phone: optional
Checkout Email Confirmation: yes
Open Registration: yes
Registration Names: required
Registration Address: required
Registration Phone: optional
Registration Voucher: required
Registration Email Confirmation: yes
Account Names: required
Account Address: required
Account Phone: optional
Account Email Confirmation: yes
Confirmation Endpoint: confirm-payment
Force Ssl Checkout: yes
Country: US
Currency: USD
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Decimals: 2
Trim Zero Decimals: yes
Recurring Payments: yes
Email From Address: service@{REDACTED}
Email From Name: Mamas Refill
Email Footer Text: {redacted}
www.{REDACTED} - service@{REDACTED}
2050 Beavercreek Rd. #101-189
Oregon City, OR 97045
Email Header Image: 19626
Cert Bg Width: 500
Cert Bg Height: 500
Cert Legacy Compat: no


Gateways
-------------------------------------------

Stripe: Enabled
Stripe Test Mode: Disabled
Stripe Logging: no
Stripe Order: 1
PayPal: Enabled
PayPal Test Mode: Disabled
PayPal Logging: no
PayPal Order: 2
Manual: Disabled
Manual Logging: no
Manual Order: 1


Server
-------------------------------------------

Mysql Version: 5.6.40
Php Curl: Yes
Php Default Timezone: UTC
Php Fsockopen: Yes
Php Max Input Vars: 3000
Php Max Upload Size: 128 MB
Php Memory Limit: 768M
Php Post Max Size: 128M
Php Soap: Yes
Php Suhosin: No
Php Time Limt: 120
Php Version: 7.3.15
Software: Apache
Wp Memory Limit: 128M


Browser
-------------------------------------------

HTTP USER AGENT: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.122 Safari/537.36


Theme
-------------------------------------------

Name: LifterLMS LaunchPad
Version: 2.4.6
Themeuri: https://lifterlms.com/launchpad/
Authoruri: https://gocodebox.com
Template:
Child Theme: No
Llms Support: No


Plugins
-------------------------------------------

Admin Custom Login: 3.1.4
Beaver Builder Plugin (Standard Version): 2.3.2.2
Caldera Forms: 1.8.10
CB Custom Beaver Builder Modules: 1.6.1
Classic Editor: 1.5
Google Analyticator: 6.5.4
index: 5.0
Jetpack by WordPress.com: 8.3
LifterLMS: 3.37.11
LifterLMS Customizations: 1.0.0
LifterLMS Gravity Forms: 2.0.1
LifterLMS Helper: 3.0.2
LifterLMS Labs: 1.5.3
LifterLMS MailChimp Add-on: 3.1.2
LifterLMS PayPal Gateway: 1.2.0
LifterLMS Stripe Payment Gateway: 5.2.0
Loco Translate: 2.3.1
MyThemeShop Theme & Plugin Updater: 2.0.17
Nav Menu Roles: 1.9.5
Ninja Beaver Lite Addons for Beaver Builder: 2.3.2
Page Links To: 3.3.3
Print-O-Matic: 1.7.12
SG Optimizer: 5.4.4
Simple 301 Redirects: 1.07
User Switching: 1.5.4
Wordfence Security: 7.4.6
WPForms Lite: 1.5.9.1
WP Login Timeout Settings: 1.1.3
WP Maintenance Mode: 2.2.4
WP Quiz Pro: 2.1.0
XCloner - Site Backup and Restore: 4.1.3


Integrations
-------------------------------------------

BbPress: No
BuddyPress: No
LifterLMS Gravity Forms: No
LifterLMS MailChimp: Yes


Template Overrides
-------------------------------------------


```

</details>

### Browser, Device, and Operating System Information

n/a

### Related User Information
HS-113467

### Considerations

The issue is pretty simple to see, we just consider that the `$stats` var, which is passed through the `wp_count_comments` filter is an object, while it might even not be so.
Consider that if it wasn't for this filter:
https://github.com/gocodebox/lifterlms/blob/3.37.11/vendor/prospress/action-scheduler/classes/ActionScheduler_wpCommentLogger.php#L225
(
https://github.com/gocodebox/lifterlms/blob/3.37.11/vendor/prospress/action-scheduler/classes/ActionScheduler_wpCommentLogger.php#L167
)
we would have this issue, without forcing, without conflicting plugins.

This might even be due to some wrong custom code, but still I think we should prevent this kind of issues by checking on the nature of the entity we're about to manipulate, if it's a wp filter param, especially if wp in a way justifies it, well actually id doesn't really, but it's ambiguous:
https://github.com/WordPress/WordPress/blob/5.3.2/wp-includes/comment.php#L1341

**Comment by thomasplevy** [comment]
@eri-trabiccolo 

> but still I think we should prevent this kind of issues by checking on the nature of the entity we're about to manipulate

Agreed.

And it's an easy enough fix. Thanks for figuring out how it could happen and putting in the effort to get some sort of reproduction steps.

**Comment by thomasplevy** [comment]
Also, this is very likely unrelated to the issue at hand but may be worth tackling while working on this issue: https://github.com/gocodebox/lifterlms/issues/901

(To whoever ends up working on this)

**Comment by eri-trabiccolo** [comment]
I was about to link this to the other issue but I was distracted by slack :P

**Comment by thomasplevy** [comment]
> Slack is incredible for my productivity!

~ A ~~real~~ Slack User

**Comment by eri-trabiccolo** [comment]
same notice here:
HS-113564

**Comment by eri-trabiccolo** [comment]
@thomasplevy 
WPForms (lite) is the shared (between the two cases) plug-in that causes this issue.
IMHO it's because they use a different (newer) version of the action scheduler library which is loaded in place of our one.
something must have changed in that library and that filter I referenced before is not called the same way as before.

**Comment by nrherron92** [comment]
Also this ticket: HS-113729

**Comment by thomasplevy** [comment]
@eri-trabiccolo 

> WPForms (lite) is the shared (between the two cases) plug-in that causes this issue.

Have you reproduced this yet?

> IMHO it's because they use a different (newer) version of the action scheduler library which is loaded in place of our one.

I haven't run tests on 3.x of the action scheduler, see #737 for related

**Comment by thomasplevy** [comment]
@nrherron92 for all affected users disabling WP_DEBUG_DISPLAY "fixes" the issue acceptably since it's a warning (not an error) it will fail silently with no major issues other than the visible error message is not pleasing. Having error reporting disabled in production is advisable anyway for a whole host of reasons.

I'll take a closer look at this today since it's probably related to a new conflict with a "supported" plugin as @eri-trabiccolo has hypothesized and try to fast track a fix into a release today or tomorrow

**Comment by eri-trabiccolo** [comment]
@thomasplevy 
yes I reproduced it yesterday.

**Comment by nrherron92** [comment]
@thomasplevy I went ahead and updated users about turning the display off!

**Comment by thomasplevy** [comment]
@nrherron92 and @eri-trabiccolo 

This is now fixed in 3.37.12 -- please notify any users waiting on an update from us
